lua/mason-registry/init.lua

```diff
@@ -193,4 +193,23 @@ function M.refresh(cb)
     end
 end
 
+---table<string, string[]>
+local aliases = {}
+
+---Register aliases for the specified packages
+---@param new_aliases table<string, string[]>
+function M.register_package_aliases(new_aliases)
+    for key, value in pairs(new_aliases) do
+        aliases[key] = aliases[key] or {}
+        for _, alias in pairs(value) do
+            table.insert(aliases[key], alias)
```

Comment:
I noticed another thing: I don't think we should apply the alias in case it correspond with the package name already as it's not really an alias that point (it also makes the UI look a bit weird with duplicate names).

```suggestion
    for pkg_name, pkg_aliases in pairs(new_aliases) do
        aliases[pkg_name] = aliases[pkg_name] or {}
        for _, alias in pairs(value) do
            if alias ~= pkg_name then
              table.insert(aliases[pkg_name], alias)
            end
```

Reply:
Nice yeah that makes sense to me. Done